### PR TITLE
Add no provisioner option to storage class form

### PR DESCRIPTION
**Pull request by pcbailey**
This PR is to add the no-provisioner option to the storage class form per https://bugzilla.redhat.com/1648828.

### CODE REVIEW

File: frontend/public/components/storage-class-form.tsx [hunk 1]
```diff
@@ -542,6 +549,11 @@ export class StorageClassForm_ extends React.Component<StorageClassFormProps, St
       data.reclaimPolicy = this.state.newStorageClass.reclaim;
     }
 
+    const volumeBindingMode = _.get(this.storageTypes[type], 'volumeBindingMode', null);
+    if (volumeBindingMode !== null) {
```
Comment (spadgett) [inline frontend/public/components/storage-class-form.tsx]: Unless there's a reason we would ever need to add values like empty string to the request, prefer

```suggestion
    if (volumeBindingMode) {
```

File: frontend/public/components/storage-class-form.tsx [hunk 2]
```diff
@@ -681,9 +693,15 @@ export class StorageClassForm_ extends React.Component<StorageClassFormProps, St
...
-    const dynamicContent = Object.keys(this.storageTypes[this.state.newStorageClass.type].parameters).map(key => {
-      const parameter = this.storageTypes[this.state.newStorageClass.type].parameters[key];
-      const validationMsg = _.get(this.state.newStorageClass.parameters[key], 'validationMsg', null);
+    const parameters = this.storageTypes[this.state.newStorageClass.type].parameters;
+
+    if (_.isEmpty(parameters)) {
+      return null;
+    }
+
+    const dynamicContent = Object.keys(parameters).map(key => {
```
Comment (spadgett) [inline frontend/public/components/storage-class-form.tsx]: Existing code, but this is better as `_.map`

```suggestion
    const dynamicContent = _.map(parameters, (parameter, key) => {
```

File: frontend/public/components/storage-class-form.tsx [hunk 3]
```diff
@@ -681,9 +693,15 @@ export class StorageClassForm_ extends React.Component<StorageClassFormProps, St
...
+      const parameter = parameters[key];
+      const validationMsg = _.get(parameters[key], 'validationMsg', null);
```
Comment (spadgett) [inline frontend/public/components/storage-class-form.tsx]:
```suggestion
      const validationMsg = _.get(parameter, 'validationMsg', null);
```

File: frontend/public/components/storage-class-form.tsx [hunk 4]
```diff
@@ -882,7 +900,8 @@ export type StorageClass = {
   metadata: object,
   provisioner: string,
   parameters: object,
-  reclaimPolicy?: string
+  reclaimPolicy?: string,
```
Comment (spadgett) [inline frontend/public/components/storage-class-form.tsx]: I missed this on last review, but the type declarations should use `;` instead of `,`

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
```
/go/src/github.com/openshift/console/frontend/public/components/storage-class-form.tsx
  49:48  error  Missing trailing comma  comma-dangle
```

**Comment by pcbailey** [comment]
> ```
> /go/src/github.com/openshift/console/frontend/public/components/storage-class-form.tsx
>   49:48  error  Missing trailing comma  comma-dangle
> ```

Thanks. I found it, too.. =) When I saw the failure, I realized I had forgotten to run the tests locally. I fixed it in the commit I pushed just before requesting a retest. Hopefully, it's good to go now.

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Pull request merged**